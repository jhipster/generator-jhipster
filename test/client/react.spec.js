const expect = require('expect');
const path = require('path');
const { skipPrettierHelpers: helpers } = require('../utils/utils');
const { OAUTH2 } = require('../../jdl/jhipster/authentication-types');
const { MICROSERVICE } = require('../../jdl/jhipster/application-types');
const { REACT } = require('../../jdl/jhipster/client-framework-types');

const commonOptions = { clientFramework: REACT };

describe('JHipster react generator', () => {
  describe('microfrontend', () => {
    it('should not succeed', async () => {
      await expect(
        helpers
          .create(path.join(__dirname, '../../generators/client'))
          .withOptions({
            skipInstall: true,
            auth: OAUTH2,
            enableTranslation: true,
            applicationType: MICROSERVICE,
            nativeLanguage: 'en',
            languages: ['fr', 'en'],
            ...commonOptions,
          })
          .run()
      ).rejects.toThrow("Client framework react doesn't support microfrontends");
    });
  });
});
