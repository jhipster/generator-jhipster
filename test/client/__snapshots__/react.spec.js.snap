// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JHipster react generator microfrontend should match generated files snapshot 1`] = `
Object {
  ".editorconfig": Object {
    "stateCleared": "modified",
  },
  ".eslintignore": Object {
    "stateCleared": "modified",
  },
  ".eslintrc.json": Object {
    "stateCleared": "modified",
  },
  ".gitattributes": Object {
    "stateCleared": "modified",
  },
  ".gitignore": Object {
    "stateCleared": "modified",
  },
  ".husky/pre-commit": Object {
    "stateCleared": "modified",
  },
  ".lintstagedrc.js": Object {
    "stateCleared": "modified",
  },
  ".prettierignore": Object {
    "stateCleared": "modified",
  },
  ".prettierrc": Object {
    "stateCleared": "modified",
  },
  ".yo-rc.json": Object {
    "stateCleared": "modified",
  },
  "README.md": Object {
    "stateCleared": "modified",
  },
  "jest.conf.js": Object {
    "stateCleared": "modified",
  },
  "package.json": Object {
    "stateCleared": "modified",
  },
  "postcss.config.js": Object {
    "stateCleared": "modified",
  },
  "sonar-project.properties": Object {
    "stateCleared": "modified",
  },
  "src/main/resources/i18n/messages_en.properties": Object {
    "stateCleared": "modified",
  },
  "src/main/resources/i18n/messages_fr.properties": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/404.html": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/WEB-INF/web.xml": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/_bootstrap-variables.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/app.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/app.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/axios-interceptor.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/axios-interceptor.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/constants.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/dayjs.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/error-middleware.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/icon-loader.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/logger-middleware.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/notification-middleware.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/notification-middleware.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/store.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/translation-middleware.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/config/translation.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/entities/menu.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/entities/reducers.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/entities/routes.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/index.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/main.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/administration.reducer.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/administration.reducer.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/configuration/configuration.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/docs/docs.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/docs/docs.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/health/health-modal.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/health/health.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/index.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/logs/logs.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/administration/metrics/metrics.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/home/home.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/home/home.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/login/login-redirect.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/modules/login/logout.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/routes.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/setup-tests.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/DurationFormat.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/auth/private-route.spec.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/auth/private-route.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/error/error-boundary-route.spec.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/error/error-boundary-route.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/error/error-boundary.spec.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/error/error-boundary.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/error/error-loading.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/error/page-not-found.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/footer/footer.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/footer/footer.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/header/header-components.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/header/header.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/header/header.spec.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/header/header.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/menus/account.spec.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/menus/account.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/menus/admin.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/menus/entities.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/menus/index.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/menus/locale.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/menus/menu-components.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/menus/menu-item.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/password/password-strength-bar.scss": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/layout/password/password-strength-bar.tsx": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/model/user.model.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/application-profile.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/application-profile.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/authentication.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/authentication.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/index.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/locale.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/locale.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/reducer.utils.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/user-management.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/reducers/user-management.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/util/date-utils.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/util/entity-utils.spec.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/util/entity-utils.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/util/pagination.constants.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/shared/util/url-utils.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/app/typings.d.ts": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/css/loading.css": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0.svg": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0_head-192.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0_head-256.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0_head-384.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_0_head-512.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1.svg": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1_head-192.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1_head-256.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1_head-384.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_1_head-512.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2.svg": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2_head-192.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2_head-256.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2_head-384.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_2_head-512.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3.svg": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3_head-192.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3_head-256.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3_head-384.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/jhipster_family_member_3_head-512.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/content/images/logo-jhipster.png": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/favicon.ico": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/activate.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/configuration.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/error.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/global.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/health.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/home.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/login.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/logs.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/metrics.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/password.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/register.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/reset.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/sessions.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/settings.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/en/user-management.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/activate.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/configuration.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/error.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/global.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/health.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/home.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/login.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/logs.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/metrics.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/password.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/register.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/reset.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/sessions.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/settings.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/i18n/fr/user-management.json": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/index.html": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/manifest.webapp": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/robots.txt": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/swagger-ui/dist/images/throbber.gif": Object {
    "stateCleared": "modified",
  },
  "src/main/webapp/swagger-ui/index.html": Object {
    "stateCleared": "modified",
  },
  "src/test/resources/i18n/messages_en.properties": Object {
    "stateCleared": "modified",
  },
  "src/test/resources/i18n/messages_fr.properties": Object {
    "stateCleared": "modified",
  },
  "tsconfig.json": Object {
    "stateCleared": "modified",
  },
  "tsconfig.test.json": Object {
    "stateCleared": "modified",
  },
  "webpack/environment.js": Object {
    "stateCleared": "modified",
  },
  "webpack/logo-jhipster.png": Object {
    "stateCleared": "modified",
  },
  "webpack/utils.js": Object {
    "stateCleared": "modified",
  },
  "webpack/webpack.common.js": Object {
    "stateCleared": "modified",
  },
  "webpack/webpack.dev.js": Object {
    "stateCleared": "modified",
  },
  "webpack/webpack.microfrontend.js": Object {
    "stateCleared": "modified",
  },
  "webpack/webpack.prod.js": Object {
    "stateCleared": "modified",
  },
}
`;
